packages/shortest/src/browser/core/browser-tool.ts

```diff
@@ -875,4 +875,84 @@ export class BrowserTool extends BaseBrowserTool {
       },
     );
   }
+
+  /**
+   * Waits for DOM to stabilize
+   * DOM is considered stabilized when:
+   * - DOMContentLoaded is fired
+   * - No mutations are detected for 1 second (e.g new elements such as modals, popups, etc.)
+   */
+  public async waitForStableDOM(): Promise<void> {
+    const page = this.page;
+    if (!page) {
+      throw new Error("No page found.");
+    }
+    try {
+      await this.waitForDOMContentLoaded();
+      await page.evaluate(async () => {
+        return await new Promise<void>((resolve) => {
+          const createTimeout = () => {
+            return setTimeout(() => {
+              resolve();
+              observer.disconnect();
+            }, 1000);
+          };
+
+          let timeout = createTimeout();
+
+          const observer = new MutationObserver(() => {
+            clearTimeout(timeout);
+            timeout = createTimeout();
+          });
+
+          observer.observe(window.document.body, {
+            childList: true,
+            subtree: true,
+          });
+        });
+      });
+    } catch (error) {
+      console.log("Failed to wait for stable DOM:", error);
+      throw error;
+    }
+  }
+
+  /**
+   * DOM is considered loaded when DOMContentLoaded is fired
+   * @see https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event
```

Comment:
Shouldn't be [`@link`](https://jsdoc.app/tags-inline-link)?

```suggestion
   * @link https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event
```